### PR TITLE
Fix mxint cast (for the last time)

### PR DESCRIPTION
**Pull request by splogdes**
This pull request includes several changes to the `mxint_cast` module in the `src/mase_components/linear_layers/mxint_operators/rtl/mxint_cast.sv` file and updates to the corresponding test file `mxint_cast_tb.py`. The changes aim to improve the handling of edge cases, enhance the accuracy of calculations, and update test parameters.

### Improvements to edge case handling and accuracy:

* Added a condition to handle the case where `log2_max_value` is zero in the computation of `edata_out` to avoid potential errors.
* Updated the calculation of `shift_value` and introduced `max_value` to ensure accurate handling of large numbers and edge cases.
* Introduced a two's complement function to handle signed arithmetic more effectively and updated conditions for `mdata_out` to use this function.

### Test parameter updates:

* Increased `MAX_MANTISSA` from 16 to 32 and adjusted the range for `in_man` and `out_man` in the `get_mxint_cast_config_random` function to allow for a wider range of test cases.
* Updated the parameter overrides in the `test_mxint_cast_random` function to reflect the new maximum values for mantissa and exponent widths.

### CODE REVIEW

File: src/mase_components/linear_layers/mxint_operators/rtl/mxint_cast.sv [hunk 1]
```diff
@@ -167,28 +188,45 @@ module mxint_cast #(
...
       else if (shift_value >= 0) mdata_out[i] = mbuffer_data_for_out[i] <<< shift_value;
 
-      else mdata_out[i] = mbuffer_data_for_out[i] >>> -shift_value;
+      else mdata_out[i] = mbuffer_data_for_out[i] >>> twos_complement(shift_value);
 
     end
 
   end
 
+  // =============================
```
Comment (luigirinaldi) [inline src/mase_components/linear_layers/mxint_operators/rtl/mxint_cast.sv:217]: crazy stuff right here

**Review by luigirinaldi — APPROVED** [review]
